networkx/algorithms/community/tests/test_modularity_max.py

```diff
@@ -146,7 +146,7 @@ def test_modularity_communities_directed_weighted():
 
     # A large weight of the edge (2, 6) causes 6 to change group, even if it shares
     # only one connection with the new group and 3 with the old one.
-    G[2][6]["weight"] = 20
+    G.add_edge(2, 6, weight=20)
```

Comment:
Nothing breaks if I leave this as is, because we disable caching when testing backends. But, I may sometimes experiment with enabling caching while testing, and if we can get it to work, it may be a nice option to add for testing (but probably not b/c of the maintenance burden).

Up to you. I can reverse these changes to the tests. I don't think there would be that many more changes to be cache-friendly though, so I thought "might as well".

Reply:
How about, for now, revert and add a line like `G.__networkx_cache__.clear()`? 
That signals that this kind of change would break caching. And when we get to testing caching in the tests, we can replace it with a test that caching would break and add the `add_edge` version that doesn't break.